flox/xarray.py

```diff
@@ -300,9 +304,9 @@ def xarray_reduce(
                     f"Please provided bin edges for group variable {idx} "
                     f"named {group_names[idx]} in expected_groups."
                 )
-            expected_groups[idx] = _get_expected_groups(b.data, sort=sort, raise_if_dask=True)
+            expected_groups[idx] = _get_expected_groups(b_.data, sort=sort, raise_if_dask=True)
 
-    expected_groups = _convert_expected_groups_to_index(expected_groups, isbin, sort=sort)
+    expected_groups = _convert_expected_groups_to_index(expected_groups, isbins, sort=sort)
     group_shape = tuple(len(e) for e in expected_groups)
```

Comment:
@dcherian, I get `error: Argument 1 to "len" has incompatible type "Optional[Any]"; expected "Sized"  [arg-type]`.
It's because `expected_groups` can return `None` which `len()` does not allow. 

I don't understand how `group_shape` is used, if `None` is returned should the loop just replace with a zero or skip?

Reply:
All `None` elements should have been removed in the previous loop:
```
if expect is None:
    ...
```